Lib/shutil.py

```diff
@@ -1449,6 +1452,17 @@ def _access_check(fn, mode):
             and not os.path.isdir(fn))
 
 
+def _win_path_needs_curdir(cmd, mode):
+    """
+    On Windows, we can use NeedCurrentDirectoryForExePath to figure out
+    if we should add the cwd to PATH when searching for executables if
+    the mode is executable.
+    """
+    return mode & os.X_OK and _winapi.NeedCurrentDirectoryForExePath(
+        cmd if isinstance(cmd, str) else os.fsdecode(cmd)
+    )
```

Comment:
`_win_path_needs_curdir()` should be true if `mode & os.X_OK` is false. Otherwise it depends on `NeedCurrentDirectoryForExePath()`.

```suggestion
    return (not (mode & os.X_OK) or
            _winapi.NeedCurrentDirectoryForExePath(
                cmd if isinstance(cmd, str) else os.fsdecode(cmd))
```

---

The `bytes` handling in this function is awkward. I prefer how `os.path.normpath()` works: decode and then encode the result. But requiring a `bytes` `cmd` to be decodable would be new behavior.